src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTracePipelineLogic.java

```diff
@@ -310,7 +312,11 @@ private Long getNanosInQueue(BuildPipelineNode current) {
             }
         }
 
-        final String gitCommit = envVars.get("GIT_COMMIT") !=  null ? envVars.get("GIT_COMMIT") : buildData.getGitCommit("");
+        // If we could detect a valid commit, the buildData object will contain that commit.
+        // If we could not detect a valid commit, that means that the GIT_COMMIT environment variable
+        // was overridden by the user at top level, so we set the content what we have (despite it's not valid).
+        // We will show a logger.warning at the end of the pipeline.
+        final String gitCommit = (isValidCommit(envVars.get("GIT_COMMIT"))) ? envVars.get("GIT_COMMIT") : buildData.getGitCommit("");
```

Comment:
nit I would copy `envVars.get("GIT_COMMIT")` to a temporary var to make this simpler